Update pack versions (#1093)
spectrocloud/librarium: content/docs/06-integrations/00-aws-efs.md

> Palette handles setting up the AWS EFS as a volume with ease when adding the PersistentVolume storage container. Palette will dynamically provision the AWS EFS storage layer for the worker node.

Author: Removed.

## Versions Supported

<Tabs>

<Tabs.TabPane tab="1.4.x" key="1.4.x">

* **1.4.9**
* **1.4.0**

## Prerequisites

- Create the Identity and Access Management (IAM) role that allows the driver to manage AWS EFS access points. See the [Introducing Amazon EFS CSI dynamic provisioning](https://aws.amazon.com/blogs/containers/introducing-efs-csi-dynamic-provisioning/) blog for information on `EFSCSIControllerIAMPolicy`.

> - Have a filesystem created and available before you provision AWS EFS to Palette.

Author: Incorporated.

| gidRangeStart    |                  | 1000                  | Starting range of the Portable Operating System Interface(POSIX) group Id to be applied for access point root directory creation (optional).                                                                                      |
| gidRangeEnd      |                     | 2000                  | End range of the POSIX group Id (optional).                                                                                                                                                                                                 |
| basePath         |                  | /base_efs             | Path under which access points for dynamic provisioning is created. <br /> If this parameter is not specified, access points are created under the root directory of the file system.                                             |

## Usage

There are two ways to add AWS EFS to Palette:

> - Add EFS as a CSI layer in AWS/EKS.

Reviewer: ```suggestion
- Add EFS as a base CSI layer in a cluster profile.
```

| basePath         |                  | /base_efs             | Path under which access points for dynamic provisioning is created. <br /> If this parameter is not specified, access points are created under the root directory of the file system.                                             |

## Usage

There are two ways to add AWS EFS to Palette:

- Add EFS as a CSI layer in AWS/EKS.

- Add EFS as an Add-on layer, which will create a new storage class using the AWS EFS file system.

### Policy Information

> 

Reviewer: Why are we showing this IAM policy? We should add an explanation.

```yaml
{
  "Version": "2012-10-17",
  "Statement": [{
    "Effect": "Allow",
    "Action": [
      "elasticfilesystem:DescribeAccessPoints",
      "elasticfilesystem:DescribeFileSystems"
    ],
    "Resource": "*"
  },
    {
      "Effect": "Allow",
      "Action": [
        "elasticfilesystem:CreateAccessPoint"
      ],
      "Resource": "*",
      "Condition": {
        "StringLike": {
          "aws:RequestTag/efs.csi.aws.com/cluster": "true"
        }
      }
    },
    {
      "Effect": "Allow",
      "Action": "elasticfilesystem:DeleteAccessPoint",
      "Resource": "*",
      "Condition": {
        "StringEquals": {
          "aws:ResourceTag/efs.csi.aws.com/cluster": "true"
        }
      }
    }
  ]
}
```

</Tabs.TabPane>
</Tabs>

# Troubleshooting 

> **QUESTION: Is information below about "Storage Class" and "PersistentVolumnClaim" really information for the Usage section?** 

Reviewer: Yes, this is usage information.

**Should the Troubleshooting section list the following points instead?**

> **Is is safe to include the commands shown or is there the possibility they could change?. If so, the doc can list the bullet points and refer users to the Amazon troubleshooting site for more details.**

Reviewer: The commands can be used by the section needs clarification and clean-up.

- **DNS server:** Confirm the cluster's Virtual Private Cloud (VPC) uses the Amazon DNS server. To verify the DNS server, log in to the worker node and issue the following command, replacing ```region``` with your AWS Region: <br />

  ```bash 
  nslookup fs-4fxxxxxx.efs.region.amazonaws.com <amazon provided DNS IP>
  <amazon provided DNS IP = VPC network(10.0.0.0) range plus two>
  ```

- **Permissions:** Verify you have "iam" mount options in the persistent volume definition when using a restrictive file system policy. In some cases, the EFS file system policy is configured to restrict mount permissions to specific IAM roles. In this case, the EFS mount helper in the EFS CSI driver requires the ```-o iam``` mount option during the mount operation. Include the **spec.mountOptions** property:<br />

  ```bash
  spec:
    mountOptions:
      - iam
  ```
- **IAM role:** Verify the Amazon EFS CSI driver controller service account associates with the correct IAM role and the IAM role has the required permissions.

  Run the following command:

  ```bash

>   kubectl describe sa efs-csi-controller-sa -n kube-system

Reviewer: ```suggestion
  kubectl describe sa efs-csi-controller-sa --namespace kube-system
```
We should always show the long-form of command to help the reader.

>   ```

Author: fixed typo.

  
  You shoud see this annotation:

  ```bash
  eks\.amazonaws\.com/role-arn"="arn:aws:iam::111122223333:role/AmazonEKS_EFS_CSI_Driver_Policy
  ```
- **Driver Pods:** Verify the EFS CSI driver pods are running. Issue the following command to display a list of controller pods and node pods running in your cluster:

  ```bash

>   kubectl get all -l app.kubernetes.io/name=aws-efs-csi-driver -n kube-system

Reviewer: ```suggestion
  kubectl get all --label app.kubernetes.io/name=aws-efs-csi-driver --namespace kube-system
```

  ```

- **File system won't mount:** Verify the EFS mount operation from the EC2 worker node where the pod is failing to mount the file system. Log in to the Amazon EKS worker node where the pod is scheduled. Then, use the EFS mount helper to try to manually mount the EFS file system to the worker node. You can run the following command to test:

  ```bash

>   sudo mount -t -efs -o tls file-system-dns-name efs-mount-point/

Reviewer: ```suggestion
  sudo mount --types -efs --options tls file-system-dns-name efs-mount-point/
```

Mounted By:  <none>

```
Possible Troubleshooting info to add:

If you get errors in your pods when mounting your Amazon EFS volume in your Amazon EKS cluster, verify you have the following:

- An Amazon EFS file system created with a mount target in each of the worker node subnets.
- A valid EFS storage class (from the GitHub website) definition using the efs.csi.aws.com provisioner.
- A valid PersistentVolumeClaim (PVC) definition and PersistentVolume definition. This isn't needed if you're using dynamic provisioning (from the GitHub website).
- The Amazon EFS CSI driver installed in the cluster.

# Terraform

```

> data "spectrocloud_registry" "container_registry" {
>   name = "Public Repo"
> }
> 
> data "spectrocloud_pack_simple" "csi-aws-efs" {
>   name    = "csi-aws-efs"
>   version = "1.4.0"
>   type = "helm"
>   registry_uid = data.spectrocloud_registry.container_registry.id
> }

Reviewer: ```suggestion
data "spectrocloud_registry" "public_registry" {
  name = "Public Repo"
}

data "spectrocloud_pack_simple" "csi-aws-efs" {
  name    = "csi-aws-efs"
  version = "1.4.0"
  type = "helm"
  registry_uid = data.spectrocloud_registry.public_registry.id
}
```

Reviewer: @ritawatson, so the current Terraform code is if you are using this Pack as an add-on to a cluster profile.

When using this Pack as a base layer, you need the following terraform code.

```
data "spectrocloud_registry" "public_registry" {
  name = "Public Repo"
}
data "spectrocloud_pack_simple" "aws-efs" {
  name    = "aws-efs"
  version = "1.4.0"
  type = "helm"
  registry_uid = data.spectrocloud_registry.public_registry.id
}
```

Reviewer: Pink is system app (storage layer), orange is addon
![CleanShot 2023-01-30 at 19 35 56@2x](https://user-images.githubusercontent.com/29551334/215648255-d33867a0-e6e9-4f9c-b9be-98e5638c314c.png)

Reviewer: I asked in a thread that I tagged you on if this is the plan.
